server/app/src/main/java/com/google/heartrate/wearos/app/bluetooth/server/BluetoothServer.java

```diff
@@ -0,0 +1,208 @@
+     *
+     * @param requestHandler gatt service request handler to register
+     */
+    public void registerGattServiceHandler(GattServiceRequestHandler requestHandler) {
+        BluetoothGattService gattService = requestHandler.getBluetoothGattService();
+        mBluetoothGattServer.addService(gattService);
+        gattRequestHandlerByServiceUuid.put(gattService.getUuid(), requestHandler);
+    }
+
+    /**
+     * Unregister given gatt service request handler from server.
+     * Remove handler's service from the {@link BluetoothGattServer}.
+     *
+     * @param requestHandler gatt service request handler to unregister
+     */
+    public void unregisterGattServiceHandler(GattServiceRequestHandler requestHandler) {
+        BluetoothGattService gattService = requestHandler.getBluetoothGattService();
+        mBluetoothGattServer.removeService(gattService);
+        gattRequestHandlerByServiceUuid.remove(gattService.getUuid());
+    }
+
+    /**
+     * Start advertising process to advertise server existence.
+     */
+    public void start() {
+        Log.v(TAG, "Starting heart rate server");
+
+        mBluetoothAdvertiser.start(gattRequestHandlerByServiceUuid.keySet());
+    }
+
+    /**
+     * Stop server interaction with all connected clients.
+     */
+    public void stop() {
+        Log.v(TAG, "Stopping heart rate server");
+
+        for (GattServiceRequestHandler requestHandler : gattRequestHandlerByServiceUuid.values()) {
+            requestHandler.onServiceRemoved();
+        }
+        mBluetoothGattServer.close();
+        mBluetoothAdvertiser.stop();
+    }
+
+    /**
+     * Register receiver which control {@link BluetoothAdvertiser} state.
+     */
+    public void registerReceiver() {
+        Log.d(TAG, "Register receiver");
+        mContext.registerReceiver(mBluetoothReceiver,
+                new IntentFilter(BluetoothAdapter.ACTION_STATE_CHANGED));
+    }
+
+    /**
+     * Unregister receiver which control {@link BluetoothAdvertiser} state.
+     */
+    public void unregisterReceiver() {
+        Log.d(TAG, "Unregister receiver");
+        mContext.unregisterReceiver(mBluetoothReceiver);
+    }
+
+    /**
+     * Send a response to a read or write request to a remote device.
+     *
+     * <p>This function must be invoked in when a remote read/write request
+     * is received by {@link BluetoothServerCallback} read/write methods.
+     *
+     * @param device remote device to send this response to
+     * @param requestId id of the request that was received with the callback
+     * @param status status of the request to be sent to the remote devices
+     * @param offset offset for partial read/write response
+     * @param value value of the attribute that was read/written
+     */
+    public void sendResponse(BluetoothDevice device, int requestId, int status, int offset, byte[] value) {
+        mBluetoothGattServer.sendResponse(device, requestId, status, offset, value);
```

Comment:
Optional: it might be good to log when sendResponse fails (i.e. returns `false`)